File: Searching/binary.cpp
#include <iostream>

using namespace std;

int binary(int A[], int l, int r, int t){
    int mid = l + (r-l)/2;

    if( t > A[mid] ){
        return binary(A,mid,r,t);
    }

    else if( t < A[mid] ){
        return binary(A,l,mid,t);
    }

    else if( t == A[mid] ){
        return mid+1;
    }

    else
    {
        return -1;
    }
    
}

int main(){
    int A[10] = {1,2,3,4,5,6,7,10,13,34};
    int term;
    cin >> term ;
    int index = binary(A,0,9,term);
    cout << index;
}

File: Searching/binary.py
def binary(A,l,r,t):
    mid = l + (r-l)//2

    if t > A[mid]:
        return binary(A,mid,l,t)

    elif t < A[mid]:
        return binary(a,f,mid,t)
    
    elif t == A[mid]:
        return mid+1
    else:
        return -1

A =  list(map(int,input().split()))
B = int(input("ENTER THE ELEMENT TO FIND : "))

A = sorted(A)
print(A)

print(binary(A,0,len(A)-1,B))
